python/lsst/analysis/tools/actions/plot/gridPlot.py

```diff
@@ -109,19 +122,23 @@ def __call__(self, data: KeyedData, **kwargs) -> PlotResultType:
                         newData = {}
                         if val not in key:
                             continue
-                        namedKey = self.panels[index].plotElement.valsKey
-                        newData[namedKey] = data[key]
                         if xList is not None:
                             namedKey = self.panels[index].plotElement.xKey
                             newData[namedKey] = data[xList[i]]
+                            if key in xList:
```

Comment:
I know that you didn't write most of this section but some comments in here would be really useful.

Reply:
Documented as best I could since I'll likely need to work on this again.  There are multiple levels of indirection that make this tool difficult to use.